### ISSUE TITLE
[BUG]: SecAgg tutorial fails - incorrect encryption factor value

### ISSUE THREAD

**Issue by pandrey-fr** [fedbiomed/fedbiomed#1128]
**TL;DR**:

I am encountering an issue while running the SecAgg tutorial (file `notebooks/tutorials/security/secure-aggregation.ipynb`), on the current version of the develop branch (latest commit: 850f73dc3f9691064cf8ba18f0e7778b41cfbc8d). Basically, SecAgg fails, due to encryption factors' verification failing.

### Issue description

I ran the indicated setup commands:
```bash
# Set up MP-SPDZ.
./scipts/fedbiomed_configure_secagg node
# Create 3 nodes and a researcher.
./scripts/fedbiomed_run create --component NODE  --name node_1.ini
./scripts/fedbiomed_run create --component NODE  --name node_2.ini
./scripts/fedbiomed_run create --component NODE  --name node_3.ini
./scripts/fedbiomed_run configuration create --component researcher
# Set up key infrastructure for SecAgg setup.
./scripts/fedbiomed_run certificate-dev-setup
```

I also added the default MNIST dataset to all three nodes.

I then ran the researcher and nodes in parallel terminals:
```bash
./scripts/fedbiomed_run node --config node_1.ini start
./scripts/fedbiomed_run node --config node_2.ini start
./scripts/fedbiomed_run node --config node_3.ini start
./scripts/fedbiomed_run researcher start
```

And launched the SecAgg tutorial notebook, of which I ran the first few cells, unmodified.

- Nodes are properly detected, and respond as expected to the dataset query.
- SecAgg setup goes well (default biprime is used ; MP-SPDZ is called, server key is created).
- First round of local training goes well (limited to 100 iterations by default).
- Encryption of parameters and encryption factors seemingly goes well, and training replies are sent.
- Parsing of training replies goes well. (_I added instructions to export them as a pickle file for further investigation._)
- Then, SecAgg aggregation fails, with the following rather-uninformative logs:

```
2024-06-10 15:32:27,232 fedbiomed INFO - Nodes that successfully reply in round 0 ['NODE_8ac4b935-2a70-4814-98d5-1bd0fb879af7', 'NODE_ecff29e6-091f-4c36-98f3-b814a176dd37', 'NODE_b3d2a524-eca1-48f6-8146-431d4e5eaed5']

2024-06-10 15:32:27,240 fedbiomed INFO - Validating secure aggregation results...

2024-06-10 15:32:27,241 fedbiomed INFO - Aggregating 3 parameters from 3 nodes.

2024-06-10 15:32:27,241 fedbiomed DEBUG - Aggregation is completed in 0.0 seconds.


--------------------
Fed-BioMed researcher stopped due to exception:
FB417: secure aggregation error: Aggregation is failed due to incorrect decryption.
--------------------
```

As indicated by the logs, and replicated when reloading training replies, the error is raised when secure-aggregating encryption factors, that do not correspond to the expected value.

### Investigating the issue

I ran the following to get an insight of what went wrong, but did not go further then verifying that the encryption factors' decrypted value is indeed mismatching the expected value.

```python
# Recover pickle-exported training replies.
import pickle

with open("training_replies.pkl", "rb") as file:
    replies = pickle.load(file)

# Re-run their refining (by `DefaultStrategy`).
model_params, weights, total_sample_size, encryption_factors = exp._node_selection_strategy.refine(
    replies, 0
)

# Aggregate encryption factors.
import functools

biprime = exp.secagg.biprime.context["context"]["biprime"]
key = exp.secagg.servkey.context["context"]["server_key"]
num_nodes = 3
crypter = exp.secagg._secagg_crypter

aggregate = functools.partial(
    crypter.aggregate,
    current_round=0,
    num_nodes=num_nodes,
    key=key,
    total_sample_size=total_sample_size,
    biprime=biprime,
    clipping_range=exp.secagg.clipping_range,
)
result = aggregate(params=list(encryption_factors.values()), num_expected_params=1)
expected = exp.secagg.train_arguments()["secagg_random"]
print(f"Got {result}, expected [{expected}].")
```
Output:
```
"Got [8.959166234321115], expected [0.794]."
```

### What now?

Maybe something went wrong in my setup? Otherwise, I think there is an issue with SecAgg.
If anyone can try to replicate, and/or give me advice as to how to setup things properly/anew, please let me know.

**Comment by pandrey-fr** [comment]
I updated to the latest version of develop, removed my `etc/` folder and re-ran the setup (using updated instructions, since the syntax for scripts has changed). The tutorial now seems to work fine.

I am baffled as to why, but at least it works.
